src/ts/components/volumeslider.ts

```diff
@@ -42,26 +43,28 @@ export class VolumeSlider extends SeekBar {
       return;
     }
 
-    let volumeChangeHandler = () => {
-      if (player.isMuted()) {
+    volumeController.onChanged.subscribe((_, args) => {
+      if (args.muted) {
         this.setPlaybackPosition(0);
       } else {
-        this.setPlaybackPosition(player.getVolume());
+        this.setPlaybackPosition(args.volume);
       }
-    };
+    });
 
-    player.addEventHandler(player.EVENT.ON_READY, volumeChangeHandler);
-    player.addEventHandler(player.EVENT.ON_VOLUME_CHANGED, volumeChangeHandler);
-    player.addEventHandler(player.EVENT.ON_MUTED, volumeChangeHandler);
-    player.addEventHandler(player.EVENT.ON_UNMUTED, volumeChangeHandler);
+    let volumeTransition: VolumeTransition;
 
+    this.onSeek.subscribe(() => {
+       volumeTransition = volumeController.startTransition();
+    });
     this.onSeekPreview.subscribeRateLimited((sender, args) => {
-      if (args.scrubbing) {
-        player.setVolume(args.position, VolumeSlider.issuerName);
+      if (args.scrubbing && volumeTransition) {
```

Comment:
The `volumeTransition` will always be resolve to true after the first time. There is no 'release'. So is it necessary to add it here?

Reply:
It's just a safety measure in case the events for some reason do not arrive in the expected order.